package NCnorm.NProto;

message TUnicodeProcessingConfig {  // Process a symbol x if
    repeated string ClassesToProcess = 1;  // ((ClassesToProcess.contains(class(x)) ||
    repeated uint32 SymbolsToProcess = 2;  // SymbolsToProcess.contains(static_cast<ui32>(x))) &&
    repeated uint32 SymbolsToIgnore = 3;  // !SymbolsToIgnore.contains(static_cast<ui32>(x)))
}

message TNormalizationConfig {
    optional bool NeedLowercase = 1 [default = true];
    optional bool NeedRemoveAttributes = 2 [default = true];
    optional bool NeedRemoveTrailingDashStartingTokens = 3 [default = false];
    optional bool NeedSplitAlphaNumericTokens = 4 [default = false];
    optional bool NeedKeepTrailingPlusSymbols = 5 [default = false];
    optional string UnicodeNormalizationType = 6 [default = "NFC"];
    optional TUnicodeProcessingConfig UnicodeSeparateConfig = 7;
    optional TUnicodeProcessingConfig UnicodeRemoveConfig = 8;
    optional TUnicodeProcessingConfig UnicodeReplaceByWhitespaceConfig = 9;
    optional uint64 MaxTokenLength = 10 [default = 512];  // 0 for unbounded
    optional uint64 MaxTokensNumber = 11 [default = 0];  // 0 for unbounded
}
